soroban-env-common/env.json

```
@@ -2403,6 +2403,19 @@
                     ],
                     "return": "Void",
                     "docs": "Authorizes sub-contract calls for the next contract call on behalf of the current contract. Every entry in the argument vector corresponds to `InvokerContractAuthEntry` contract type that authorizes a tree of `require_auth` calls on behalf of the current contract. The entries must not contain any authorizations for the direct contract call, i.e. if current contract needs to call contract function F1 that calls function F2 both of which require auth, only F2 should be present in `auth_entries`."
+                },
+                {
+                    "export": "4",
+                    "name": "get_address_executable",
+                    "args": [
+                        {
+                            "name": "address",
+                            "type": "AddressObject"
+                        }
+                    ],
+                    "return": "Val",
+                    "docs": "Returns the executable corresponding to the provided address. When the address does not exist on-chain, returns `Void` value. When it does exist, returns a value of `AddressExecutable` contract type. It is an enum with `Wasm` value and the corresponding Wasm hash for the Wasm contracts, `StellarAsset` value for Stellar Asset contract instances, and `Account` value for the 'classic' (G-) accounts.",
```

"StellarAsset" feels pretty odd given that we don't have StellarAccount for the second item. Is there a reason we can't just call the asset contract the asset contract?
```suggestion
                    "docs": "Returns the executable corresponding to the provided address. When the address does not exist on-chain, returns `Void` value. When it does exist, returns a value of `AddressExecutable` contract type. It is an enum with `Wasm` value and the corresponding Wasm hash for the Wasm contracts, `Asset` value for Stellar Asset contract instances, and `Account` value for the 'classic' (G-) accounts.",
```

---

I'm not really comfortable changing this, give that `ContractExecutable` enum that already exists in the protocol uses `StellarAsset` (both in the XDR and in the host). Also we refer to SAC as 'Stellar Asset Contract' everywhere, so I'm not sure if just 'Asset' is sufficient to disambiguate (but the main reason is really consistency).

---

That's fair enough. The only other place in the API would be in the auth framework API, is that right? Seems like a pretty small surface area.

---

Well, it appears in the XDR as well (like `ScContractInstance` etc.). I'm not sure it's worth having divergent terminology in different places, even though the current name might be a bit unfortunate (though as I've mentioned above, I don't think it's too bad, given that SAC is a well-known term).